pages/activity.tsx

```diff
@@ -0,0 +1,121 @@
+import 'core-js/full/array/from-async';
+
+import { observer } from 'mobx-react';
+import { InferGetStaticPropsType } from 'next';
+import { FC } from 'react';
+import { Container } from 'react-bootstrap';
+
+import { PageHead } from '../components/Layout/PageHead';
+import { ArticleMeta, pageListOf, traverseTree } from './api/core';
+
+export const getStaticProps = async () => {
+  try {
+    // Try to load activities, but handle case where directory doesn't exist
+    let activities: ArticleMeta[] = [];
+
+    try {
+      const activitiesData = await Array.fromAsync(pageListOf('/article/Wiki/_posts/Activity'));
+
+      activities = activitiesData
+        .map(root => [...traverseTree(root, 'subs')])
+        .flat()
+        .filter((event): event is ArticleMeta => 'meta' in event)
+        .sort((a, b) => {
+          const dateA = a.meta?.date ? new Date(a.meta.date).getTime() : 0;
+          const dateB = b.meta?.date ? new Date(b.meta.date).getTime() : 0;
+
+          return dateB - dateA;
+        });
+    } catch (dirError) {
+      console.warn('Activity directory not found, using empty activities list:', dirError);
+    }
+
+    return {
+      props: { upcomingEvents: activities },
+      revalidate: 3600,
+    };
+  } catch (error) {
+    console.error('Error loading activities:', error);
+
+    return {
+      props: { upcomingEvents: [] },
+      revalidate: 3600,
+    };
+  }
```

Comment:
```suggestion
    return {
      props: { upcomingEvents },
      revalidate: 3600,
    };
```
让错误自然抛出，提醒开发者不能缺少这一目录，也不能在构建时忽略本地文件错误。